### ISSUE TITLE
Problem in local vs. global batch sizes?

### ISSUE THREAD

**Issue by sayakpaul** [microsoft/DeepSpeed#1115]
My current setup consists of 4 local V100s. I wanted to be able to take the advantage of distributed training and mixed-precision. The following is my `ds_report`:

```
--------------------------------------------------
DeepSpeed C++/CUDA extension op report
--------------------------------------------------
NOTE: Ops not installed will be just-in-time (JIT) compiled at
      runtime if needed. Op compatibility means that your system
      meet the required dependencies to JIT install the op.
--------------------------------------------------
JIT compiled ops requires ninja
ninja .................. [OKAY]
--------------------------------------------------
op name ................ installed .. compatible
--------------------------------------------------
cpu_adam ............... [NO] ....... [OKAY]
fused_adam ............. [NO] ....... [OKAY]
fused_lamb ............. [NO] ....... [OKAY]
 [WARNING]  sparse_attn requires one of the following commands '['llvm-config', 'llvm-config-9']', but it does not exist!
 [WARNING]  sparse_attn requires CUDA version 10.1+, does not currently support >=11 or <10.1
sparse_attn ............ [NO] ....... [NO]
transformer ............ [NO] ....... [OKAY]
stochastic_transformer . [NO] ....... [OKAY]
async_io ............... [NO] ....... [OKAY]
transformer_inference .. [NO] ....... [OKAY]
utils .................. [NO] ....... [OKAY]
quantizer .............. [NO] ....... [OKAY]
--------------------------------------------------
DeepSpeed general environment info:
torch install path ............... ['/home/jupyter/.local/lib/python3.7/site-packages/torch']
torch version .................... 1.7.1+cu110
torch cuda version ............... 11.0
nvcc version ..................... 11.0
deepspeed install path ........... ['/opt/conda/lib/python3.7/site-packages/deepspeed']
deepspeed info ................... 0.4.0+11e94e6, 11e94e6, master
deepspeed wheel compiled w. ...... torch 1.7, cuda 11.0
```

Referring to the [official CIFAR10 example](https://github.com/microsoft/DeepSpeedExamples/blob/master/cifar/cifar10_deepspeed.py), I set my training and configuration up. The number of total training samples in my dataset is 33405 (consists of 256x256x1 grayscale images) and I am setting a batch size of 256. With this, the number of batches in each epoch is supposed to be 66 (taking the ceiling). But my training logs show otherwise -- 8352. Why is that? 

I can't share the dataset yet but I am attaching my configuration and training script. Apologies in advance for the untidy training script. 

[Archive.zip](https://github.com/microsoft/DeepSpeed/files/6564056/Archive.zip)

**Comment by tjruwase** [comment]
@sayakpaul, how are you setting the batch size? Can you share your command line and json config?

**Comment by sayakpaul** [comment]
It's there in the [archive](https://github.com/microsoft/DeepSpeed/files/6564056/Archive.zip), @tjruwase.

**Comment by tjruwase** [comment]
@sayakpaul, please note that it is a lot easier if debugging clues like command line, stdout, logs are shared directly (copy-paste) into the chat. Also, can you share output of `ds_report`?

**Comment by sayakpaul** [comment]
You can see the `ds_report` in the issue description itself. Not really sure which parts I missed out on.

**Comment by tjruwase** [comment]
Yes, sorry, I saw the `ds_report` after asking. 

I took a look at the archive, it contains your script and jscon config, but this is not sufficient to understand what is going on. In particular, it is not clear which part of your log is showing the unexpected global vs local batch sizes. Sometimes the training script and deepspeed may have conflicting views of batch sizes. Sharing your command line and log will be debug further. Thanks.
